### PR TITLE
chore: update subtractions edit modal

### PR DESCRIPTION
**Pull request by raastinn**
Converting the `onUpdate()` to react-query required me to change the Detail component as well. Left that off for the overhaul of the detail component issue.

### CODE REVIEW

File: src/js/subtraction/components/Detail/__tests__/__snapshots__/EditSubtraction.test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+  label="Edit Subtraction"
+  onHide={[MockFunction spy]}
+  show={true}
+  size="sm"
+>
+  <ModalHeader>
+    Edit Subtraction
+  </ModalHeader>
+  <form
+    onSubmit={[Function]}
```
Comment (ReeceHoffmann) [inline src/js/subtraction/components/Detail/__tests__/__snapshots__/EditSubtraction.test.tsx.snap]: Are these being used? looks like you did an RTL conversion for `EditSubtracton.test.tsx` so guessing these are vestigial

File: src/js/subtraction/components/Detail/EditSubtraction.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                </ModalFooter>
+            </form>
+        </Modal>
+    );
+}
+
+export function mapDispatchToProps(dispatch) {
+    return {
+        onUpdate: (id, name, nickname) => {
+            dispatch(editSubtraction(id, name, nickname));
```
Comment (ReeceHoffmann) [inline src/js/subtraction/components/Detail/EditSubtraction.tsx:71]: No luck with the websockets?

File: src/js/subtraction/components/Detail/EditSubtraction.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                    <InputGroup>
+                        <InputLabel>Nickname</InputLabel>
+                        <Input aria-label="nickname" name="nickname" value={values.nickname} onChange={handleChange} />
+                    </InputGroup>
+                </ModalBody>
+
+                <ModalFooter>
+                    <SaveButton />
+                </ModalFooter>
+            </form>
```
Comment (ReeceHoffmann) [inline src/js/subtraction/components/Detail/EditSubtraction.tsx]: The scope of this PR is fairly small so I think it is fair here to convert this over to react-hook-form

**Review by ReeceHoffmann — CHANGES_REQUESTED** [review]
Changes so far look good. Might be a couple more things you can do. Not sure if react-query is in the cards here, might need to be done with the main component

File: src/js/subtraction/components/Detail/EditSubtraction.tsx (hunk 2 above)
Comment (raastinn) [reply src/js/subtraction/components/Detail/EditSubtraction.tsx:71]: Unfortunately not

**Review by ReeceHoffmann — APPROVED** [review]

**Pull request merged**